code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -964,8 +964,7 @@ var/global/list/damage_icon_parts = list()
 			standing = image("icon" = 'icons/mob/suit.dmi', "icon_state" = "[wear_suit.icon_state]")
 
 
-		if( istype(wear_suit, /obj/item/clothing/suit/straight_jacket) )
-			unEquip(handcuffed)

```

Comment:
Good lord, this is dumb. I never noticed it unequips on updating icons.

You could call this more of a fix, since you can handcuff people in straightjackets as of right now.